### PR TITLE
[Sharktank] LLM Chunked Scheduler Unit Tests

### PR DESCRIPTION
**Pull request by stbaione**
Stacked on #2454 
- Unit tests for llm_utils components related to `chunked prefill` flow
- Not completely exhaustive for all components, but covers the ones that I messed with to enable `chunked prefill`, specifically:
    - `BasicScheduler`
    - `ChunkScheduler`
    - `DecodeTask`
    - `PrefillTask`
    - `LlmRunner`

### CODE REVIEW

File: sharktank/tests/utils/llm_decode_tasks_test.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+            batch_size=self._batch_size,
+            block_stride=self._block_stride,
+            decode_topk_logits=None,
+        )
+
+        cache_state = [numpy.zeros((16, 256), dtype=numpy.float16)]
+        logits, indices = decode_task.run(*cache_state)
+
+        expected_tokens = numpy.array(
+            [[request[-1]] for request in requests], dtype=numpy.int64
```
[the remark this answers is not in the document]
Comment (stbaione) [reply sharktank/tests/utils/llm_decode_tasks_test.py]: Created `_int_dtype` arg
